functions/console_snippets/http/http_test.go

```
@@ -0,0 +1,72 @@
+package p
+
+import (
+	"bufio"
+	"bytes"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestHelloWorld(t *testing.T) {
+	tests := []struct {
+		name string
+		want string
+		data string
+	}{
+		{
+			name: "valid",
+			want: "Greetings, Ocean!",
+			data: `{"message": "Greetings, Ocean!"}`,
+		},
+		{
+			name: "empty",
+			want: "Hello, World!",
```

This should probably also raise the error invalid JSON.

The data seen ("") is invalid JSON and should err.